components/PronunciationChecker/index.tsx

```
@@ -0,0 +1,207 @@
+
+  // Kiểm tra hỗ trợ nhận dạng giọng nói
+  if (!browserSupportsSpeechRecognition) {
+    return (
+      <div className="p-4 bg-red-100 text-red-700 rounded">
+        Trình duyệt của bạn không hỗ trợ nhận dạng giọng nói.
+      </div>
+    );
+  }
+
+  // Xử lý khi transcript thay đổi
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)